### PR TITLE
[Refactor] Refactor debug code + mini perf enhancement.

### PR DESCRIPTION
**Pull request by bashimao**
This change contains
* Rename `.h` files to `.hpp` to avoid confusion with c std lib.
* Move basic debug structures to own header (`debug.hpp`) that can be imported with non-`nvcc` compiler.
* Small performance enhancement. `MERLIN_CHECK` will now avoid always casting to `std::string`, which saves a memory allocation.
* To improve overall syntax, changed debug macro, so that `;` is no longer optional.

### CODE REVIEW

**Comment by rhdong** [comment]
Hi @Lifann , this PR will change the filename of `merlin_localfile.h` to `merlin_localfile.hpp` , please comment here if you want, thanks!

File: include/merlin/debug.hpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
```
Comment (rhdong) [inline include/merlin/debug.hpp:1]: Maybe `debug_utils.hpp` would be better, for avoiding some name conflict, debug is too common name.

**Comment by rhdong** [comment]
It would be better to add the appropriate prefix for the git commit title：

- feat: (new feature for the user, not a new feature for build script)
- fix: (bug fix for the user, not a fix to a build script)
- docs: (changes to the documentation)
- style: (formatting, missing semi colons, etc; no production code change)
- refactor: (refactoring production code, eg. renaming a variable)
- test: (adding missing tests, refactoring tests; no production code change)
- chore: (updating grunt tasks etc; no production code change)

**Review by Lifann — APPROVED** [review]
LGTM

**Pull request merged**